### PR TITLE
Linters: add pre-commit hooks, flake8 and black

### PR DESCRIPTION
**Pull request by vladsaveliev**
Compared to interview-exercises, added few more hooks from [the list](https://github.com/pre-commit/pre-commit-hooks#hooks-available) (it's actually an exciting list, feel free to recommend more from there!), and re-enabled few inspections that didn't work for the interview-exercises repository specifically (i.g. duplicated code checks).

Updated the documentation.

### CODE REVIEW

**Comment by lgruen** [comment]
`pictures` or `figures`? ;)

File: python_code_style.md [hunk 1]
```diff
@@ -39,39 +85,21 @@ wget https://raw.githubusercontent.com/populationgenomics/team-docs/main/pylint\
...
-## PyCharm
+### PyCharm
 
-1. Make sure you have pylint installed with pip or conda.
+1. Make sure you have pylint installed with `pip install pylint pylint-quotes`.
 
-2. Install a PyCharm "Pylint" plugin ([Preferences] > [Plugins] > search for
+2. Install the PyCharm "Pylint" plugin ([Preferences] > [Plugins] > search for
    "Pylint"). It will automatically find the pylint executable. A new tool
    window called Pylint will be added. You can use it to scan the project:
```
Comment (lgruen) [inline python_code_style.md:105]: GitHub doesn't let me comment on line 112, so I'm adding a drive-by nit here ;).

Could you put the open parens for `(Cmd+K)` on the same line to avoid the extra space, please?

**Review by lgruen — APPROVED** [review]

File: python_code_style.md (hunk 1 above)
Comment (vladsaveliev) [reply python_code_style.md:105]: Sigh, automatic hard wraps aren't smart enough :)
